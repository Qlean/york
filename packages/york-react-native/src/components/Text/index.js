import React from 'react'
import PropTypes from 'prop-types'
import { Text as NativeText, StyleSheet, Platform } from 'react-native'
import { colors } from '@qlean/york-core'

import { fontFamily, fontFamilyBold } from 'york-react-native/utils/styles'

const getFontFamily = isBold =>
  Platform.select({
    ios: { fontFamily: isBold ? fontFamilyBold : fontFamily },
    android: { fontFamily: isBold ? fontFamilyBold : fontFamily },
    web: { fontFamily: 'Museo Sans', fontWeight: isBold ? '700' : '500' },
  })

const presets = {
  header1: {
    ...getFontFamily(true),
    fontSize: 25,
    lineHeight: 30,
  },
  header2: {
    ...getFontFamily(true),
    fontSize: 20,
    lineHeight: 25,
  },
  header3: {
    ...getFontFamily(true),
    fontSize: 16,
    lineHeight: 22,
  },
  text: {
    ...getFontFamily(false),
    fontSize: 16,
    lineHeight: 25,
  },
  caption: {
    ...getFontFamily(false),
    fontSize: 14,
    lineHeight: 20,
  },
  captionSmall: {
    ...getFontFamily(false),
    fontSize: 12,
    lineHeight: 20,
  },
}

const styles = StyleSheet.create({
  ...presets,
})

/**
 * Компонент для оформления текста, использует шрифт Museo Sans. Чтобы он работал, нужно добавить
 * в сборку начертания `MuseoSansCyrl-500` и `MuseoSansCyrl-700`.
 */
const Text = ({ preset, color, style: extraStyle, ...rest }) => {
  return (
    <NativeText
      style={[styles[preset], { color: colors[color] }, extraStyle]}
      {...rest}
    />
  )
}

Text.defaultProps = {
  color: 'coal',
  preset: 'text',
  style: null,
}

Text.propTypes = {
  /** Пресет, устанавливает размер, межстрочный интервал, вес и другие стилевые параметры текста */
  preset: PropTypes.oneOf(Object.keys(presets)),
  /** Цвет текста */
  color: PropTypes.oneOf(Object.keys(colors)),
  /** Дополнительные стили */
  style: NativeText.propTypes.style,
}

export default Text
