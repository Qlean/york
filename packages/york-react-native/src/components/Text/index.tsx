import React, { ReactNode } from 'react'
import {
  Text as NativeText,
  StyleSheet,
  Platform,
  TextProps,
} from 'react-native'
import { colors, colorNames } from '@qlean/york-core'

import { fontFamily, fontFamilyBold } from 'york-react-native/utils/styles'

type TextPreset = keyof typeof presets

type FontFamily = {
  fontFamily?: string
  fontWeight?: '700' | '500'
}

const getFontFamily = (isBold: boolean) =>
  Platform.select<FontFamily>({
    ios: { fontFamily: isBold ? fontFamilyBold : fontFamily },
    android: { fontFamily: isBold ? fontFamilyBold : fontFamily },
    web: { fontFamily: 'Museo Sans', fontWeight: isBold ? '700' : '500' },
  })

const presets = {
  header1: {
    ...getFontFamily(true),
    fontSize: 25,
    lineHeight: 30,
  },
  header2: {
    ...getFontFamily(true),
    fontSize: 20,
    lineHeight: 25,
  },
  header3: {
    ...getFontFamily(true),
    fontSize: 16,
    lineHeight: 22,
  },
  text: {
    ...getFontFamily(false),
    fontSize: 16,
    lineHeight: 25,
  },
  caption: {
    ...getFontFamily(false),
    fontSize: 14,
    lineHeight: 20,
  },
  captionSmall: {
    ...getFontFamily(false),
    fontSize: 12,
    lineHeight: 15,
  },
}

const styles = StyleSheet.create(presets)

type Props = {
  /** Пресет, устанавливает размер, межстрочный интервал, вес и другие стилевые параметры текста */
  preset?: TextPreset
  /** Цвет текста */
  color?: colorNames
  children: ReactNode
} & TextProps

/**
 * Компонент для оформления текста, использует шрифт Museo Sans. Чтобы он работал, нужно добавить
 * в сборку начертания `MuseoSansCyrl-500` и `MuseoSansCyrl-700`.
 */
const Text = ({
  preset = 'text',
  color = colorNames.coal,
  style,
  ...rest
}: Props) => (
  <NativeText
    style={[styles[preset], { color: colors[color] }, style]}
    {...rest}
  />
)

Text.presets = presets

export default Text
