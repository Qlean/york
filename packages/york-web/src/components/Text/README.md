```js
import { Separator } from '@qlean/york-web'
const presets = [
  {
    title: 'Header 1',
    caption: 'Заголовок страниц десктопной версии',
    preset: 'header1',
  },
  {
    title: 'Header 2',
    caption: 'Заголовок блоков страниц в десктопной версии',
    preset: 'header2',
  },
  {
    title: 'Header 3',
    caption:
      'Заголовок страницы в мобильной версии, выделение контента внутри блока на десктопе',
    preset: 'header3',
  },
  {
    title: 'Header 4',
    caption:
      'Заголовок блока в мобильной версии, название секций в формах и личном кабинете',
    preset: 'header4',
  },
  {
    title: 'Header 5',
    caption: 'Заголовок страницы в десктопной версии',
    preset: 'header5',
  },
  {
    title: 'Text Large',
    caption: 'Текст на лендингах',
    preset: 'textLarge',
  },
  {
    title: 'Text Strong',
    preset: 'textStrong',
  },
  {
    title: 'Text',
    caption: 'Обычный текст',
    preset: 'text',
  },
  {
    title: 'Caption',
    preset: 'caption',
  },
]

;<>
  {presets.map((text, index) => (
    <div key={index}>
      <Text htmlTag="div" preset="text">
        {text.title}
      </Text>
      <Text htmlTag="div" color="grey"preset="text">
        {text.caption}
      <Text htmlTag="div" preset={text.preset}>
        Шнауцеры — крепкие, довольно коренастые собаки квадратного формата с чуть наклонной спиной
      </Text>
      </Text>
      <Separator height={8} />
    </div>
  ))}
</>
```
