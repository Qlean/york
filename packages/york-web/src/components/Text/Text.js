import PropTypes from 'prop-types';
import styled from 'styled-components';
import { colors } from 'york-core';
import fonts from '../../styles/fonts';

const Text = styled.span`
  ${({ fontWeight }) => fonts[fontWeight]}
  font-size: ${({ fontSize }) => `${fontSize}px`};
  line-height: ${({ lineHeight }) => `${lineHeight}px`};
  color: ${({ color }) => colors[color] || color};
`;

Text.defaultProps = {
  fontWeight: 'normal',
  fontSize: 16,
  lineHeight: 25,
  color: 'coal',
};

Text.propTypes = {
  fontWeight: PropTypes.oneOf(['bold', 'normal', 'light']),
  fontSize: PropTypes.number,
  lineHeight: PropTypes.number,
  color: PropTypes.string,
};

export default Text;
