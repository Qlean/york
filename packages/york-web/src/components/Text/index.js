import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import * as R from 'ramda'
import { colors } from '@qlean/york-core'

import {
  media,
  unwrapResponsivePreset,
  unwrapResponsiveProps,
  mergeStyleProps,
  getResponsivePropTypes,
} from 'styles'

const htmlTags = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'div', 'span']

const presets = {
  header1: {
    fontSize: 70,
    lineHeight: 75,
    fontWeight: 700,
  },
  header2: {
    fontSize: 40,
    lineHeight: 45,
    fontWeight: 700,
  },
  header3: {
    fontSize: 30,
    lineHeight: 35,
    fontWeight: 700,
  },
  header4: {
    fontSize: 25,
    lineHeight: 30,
    fontWeight: 700,
  },
  header5: {
    fontSize: 20,
    lineHeight: 25,
    fontWeight: 700,
  },
  textStrong: {
    fontSize: 16,
    lineHeight: 25,
    fontWeight: 700,
  },
  textLarge: {
    fontSize: 20,
    lineHeight: 30,
    fontWeight: 500,
  },
  text: {
    fontSize: 16,
    lineHeight: 25,
    fontWeight: 500,
  },
  link: {
    fontSize: 15,
    lineHeight: 25,
    fontWeight: 700,
    letterSpacing: 1,
    textTransform: 'uppercase',
  },
  caption: {
    fontSize: 14,
    lineHeight: 20,
    fontWeight: 500,
  },
  captionSmall: {
    fontSize: 12,
    lineHeight: 15,
    fontWeight: 500,
  },
}

const getBaseCss = ({
  color,
  fontWeight,
  fontSize,
  lineHeight,
  textAlign,
  fontStyle,
  letterSpacing,
  textTransform,
  textDecoration,
}) => `
  color: ${color === 'inherit' ? 'inherit' : colors[color]};
  font-weight: ${fontWeight};
  font-size: ${fontSize}px;
  line-height: ${lineHeight}px;
  ${textAlign ? `text-align: ${textAlign}` : ''};
  ${fontStyle ? `font-style: ${fontStyle}` : ''};
  ${letterSpacing ? `letter-spacing: ${letterSpacing}px` : ''};
  ${textTransform ? `text-transform: ${textTransform}` : ''};
  ${textDecoration ? `text-decoration: ${textDecoration}` : ''};
`

const defaultProps = {
  preset: 'text',
  color: 'coal',
}

const getCss = initialProps => {
  const props = { ...defaultProps, ...initialProps }
  const { mobileProps, baseProps, wideProps } = mergeStyleProps([
    unwrapResponsivePreset('preset', presets, props),
    unwrapResponsiveProps(['color'], props),
  ])
  return `
    font-family: "Museo Sans";
    margin: 0;
    ${media.mobile(getBaseCss(mobileProps))}
    ${media.base(getBaseCss(baseProps))}
    ${media.wide(getBaseCss(wideProps))}
  `
}

const StyledText = styled.span`
  ${getCss}
`

const components = R.pipe(
  R.map(tag => [tag, StyledText.withComponent(tag)]),
  R.fromPairs,
)(htmlTags)

export default function Text({ htmlTag, ...rest }) {
  const StyledTextComponent = components[htmlTag]
  return <StyledTextComponent {...rest} />
}

const propTypes = {
  /** Пресет */
  preset: PropTypes.oneOf(Object.keys(presets)),
  /** Цвет текста */
  color: PropTypes.oneOf([...Object.keys(colors), 'inherit']),
}

Text.propTypes = {
  /** HTML-тег, который будет использован как обёртка для текста */
  htmlTag: PropTypes.oneOf(htmlTags),
  ...propTypes,
  ...getResponsivePropTypes(propTypes),
}

Text.defaultProps = {
  htmlTag: 'span',
  ...defaultProps,
}
