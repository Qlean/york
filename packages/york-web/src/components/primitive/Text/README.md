```js
const presets = [
  {
    caption: 'Заголовок страниц десктопной версии',
    preset: 'header1',
  },
  {
    caption: 'Заголовок блоков страниц в десктопной версии',
    preset: 'header2',
  },
  {
    caption:
      'Заголовок страницы в мобильной версии, выделение контента внутри блока на десктопе',
    preset: 'header3',
  },
  {
    caption:
      'Заголовок блока в мобильной версии, название секций в формах и личном кабинете',
    preset: 'header4',
  },
  {
    caption: 'Заголовок страницы в десктопной версии',
    preset: 'header5',
  },
  {
    caption: 'Текст на лендингах',
    preset: 'textLarge',
  },
  {
    preset: 'textStrong',
  },
  {
    caption: 'Обычный текст',
    preset: 'text',
  },
  {
    caption: 'Пункты в меню, некоторые ссылки',
    preset: 'link',
  },
  {
    caption: 'Подписи в кнопках, тултипы',
    preset: 'caption',
  },
  {
    caption: 'Пояснение под кнопками, юридический текст',
    preset: 'captionSmall',
  },
]

;<Example.Showcase overflow="hidden">
  {presets.map(({ preset, caption }) => (
    <Example.ShowcaseItem key={preset} title={preset} caption={caption}>
      <Text preset={preset}>{Example.text.medium}</Text>
    </Example.ShowcaseItem>
  ))}
</Example.Showcase>
```
