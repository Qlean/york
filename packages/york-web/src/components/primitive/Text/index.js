import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { colors } from '@qlean/york-core'

import {
  fontFamily,
  media,
  normalizeResponsivePreset,
  normalizeResponsiveProps,
  mergeStyleProps,
  getResponsivePropTypes,
} from 'york-web/utils'

const presets = {
  header1: {
    fontSize: 50,
    lineHeight: 55,
    fontWeight: 700,
  },
  header2: {
    fontSize: 40,
    lineHeight: 45,
    fontWeight: 700,
  },
  header3: {
    fontSize: 30,
    lineHeight: 35,
    fontWeight: 700,
  },
  header4: {
    fontSize: 25,
    lineHeight: 30,
    fontWeight: 700,
  },
  header5: {
    fontSize: 20,
    lineHeight: 25,
    fontWeight: 700,
  },
  textStrong: {
    fontSize: 16,
    lineHeight: 25,
    fontWeight: 700,
  },
  textLarge: {
    fontSize: 20,
    lineHeight: 30,
    fontWeight: 500,
  },
  text: {
    fontSize: 16,
    lineHeight: 25,
    fontWeight: 500,
  },
  link: {
    fontSize: 15,
    lineHeight: 25,
    fontWeight: 700,
    letterSpacing: 1,
    textTransform: 'uppercase',
  },
  caption: {
    fontSize: 14,
    lineHeight: 20,
    fontWeight: 500,
  },
  captionSmall: {
    fontSize: 12,
    lineHeight: 15,
    fontWeight: 500,
  },
}

const getBaseCss = ({
  color,
  fontWeight,
  fontSize,
  lineHeight,
  letterSpacing,
}) => `
  font-size: ${fontSize}px;
  color: ${color === 'inherit' ? 'inherit' : colors[color]};
  font-weight: ${fontWeight};
  line-height: ${lineHeight}px;
  ${letterSpacing ? `letter-spacing: ${letterSpacing}px` : ''};
`

const defaultProps = {
  preset: 'text',
  color: 'coal',
}

const getCss = initialProps => {
  const props = { ...defaultProps, ...initialProps }
  const { mobileProps, baseProps, wideProps } = mergeStyleProps([
    normalizeResponsivePreset(({ preset }) => preset, presets, props),
    normalizeResponsiveProps(['color'], props),
  ])
  return `
    font-family: ${fontFamily};
    margin: 0;
    ${media.mobile(getBaseCss(mobileProps))}
    ${media.base(getBaseCss(baseProps))}
    ${media.wide(getBaseCss(wideProps))}
  `
}

const StyledText = styled.span`
  ${getCss}
`

/**
 * Компонент для оформления текста, использует шрифт Museo Sans.
 */
const Text = props => {
  return <StyledText {...props} />
}

const propTypes = {
  /** Пресет, устанавливает размер, межстрочный интервал, вес и другие стилевые параметры текста */
  preset: PropTypes.oneOf(Object.keys(presets)),
  /** Цвет текста */
  color: PropTypes.oneOf([...Object.keys(colors), 'inherit']),
}

Text.defaultProps = defaultProps

Text.propTypes = {
  ...propTypes,
  ...getResponsivePropTypes(propTypes),
}

export default Text
