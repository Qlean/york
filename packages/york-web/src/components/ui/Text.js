import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import R from 'ramda';
import { colors } from '@qlean/york-core';

import {
  media,
  unwrapResponsivePreset,
  unwrapResponsiveProps,
  mergeStyleProps,
  getResponsivePropTypes,
} from '../../utils/styles';

export const htmlTags = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'div',
  'span',
];

export const presets = {
  ph1: {
    fontSize: 70,
    lineHeight: 75,
    fontWeight: 700,
  },
  ph2: {
    fontSize: 60,
    lineHeight: 65,
    fontWeight: 700,
  },
  ph3: {
    fontSize: 50,
    lineHeight: 55,
    fontWeight: 700,
  },
  h1: {
    fontSize: 40,
    lineHeight: 45,
    fontWeight: 700,
  },
  h2: {
    fontSize: 30,
    lineHeight: 35,
    fontWeight: 700,
  },
  h3: {
    fontSize: 25,
    lineHeight: 30,
    fontWeight: 700,
  },
  h4: {
    fontSize: 20,
    lineHeight: 25,
    fontWeight: 700,
  },
  h5: {
    fontSize: 16,
    lineHeight: 25,
    fontWeight: 700,
  },
  text1: {
    fontSize: 20,
    lineHeight: 30,
    fontWeight: 500,
  },
  text2: {
    fontSize: 16,
    lineHeight: 25,
    fontWeight: 500,
  },
  link: {
    fontSize: 15,
    lineHeight: 25,
    fontWeight: 700,
    letterSpacing: 1,
    textTransform: 'uppercase',
  },
  caption1: {
    fontSize: 14,
    lineHeight: 20,
    fontWeight: 500,
  },
  caption2: {
    fontSize: 12,
    lineHeight: 15,
    fontWeight: 500,
  },
};

const getBaseCss = ({
  color,
  fontWeight,
  fontSize,
  lineHeight,
  textAlign,
  fontStyle,
  letterSpacing,
  textTransform,
  textDecoration,
}) => `
  color: ${color === 'inherit' ? 'inherit' : colors[color]};
  font-weight: ${fontWeight};
  font-size: ${fontSize}px;
  line-height: ${lineHeight}px;
  ${textAlign ? `text-align: ${textAlign}` : ''};
  ${fontStyle ? `font-style: ${fontStyle}` : ''};
  ${letterSpacing ? `letter-spacing: ${letterSpacing}px` : ''};
  ${textTransform ? `text-transform: ${textTransform}` : ''};
  ${textDecoration ? `text-decoration: ${textDecoration}` : ''};
`;

const defaultProps = {
  preset: 'text2',
  color: 'coal',
};

export const getCss = (initialProps) => {
  const props = { ...defaultProps, ...initialProps };
  const {
    mobileProps,
    baseProps,
    wideProps,
  } = mergeStyleProps([
    unwrapResponsivePreset('preset', presets, props),
    unwrapResponsiveProps([
      'color',
      'fontWeight',
      'fontSize',
      'textAlign',
      'lineHeight',
      'textDecoration',
      'textTransform',
    ], props),
  ]);
  return `
    font-family: "Museo Sans";
    margin: 0;
    ${media.mobile(getBaseCss(mobileProps))}
    ${media.base(getBaseCss(baseProps))}
    ${media.wide(getBaseCss(wideProps))}
  `;
};

const StyledText = styled.span`
  ${getCss}
`;

const components = R.pipe(
  R.map(tag => [tag, StyledText.withComponent(tag)]),
  R.fromPairs,
)(htmlTags);

export default function Text({ htmlTag, ...rest }) {
  const StyledTextComponent = components[htmlTag];
  return <StyledTextComponent {...rest}/>;
}

Text.propTypes = {
  htmlTag: PropTypes.oneOf(htmlTags),
  ...getResponsivePropTypes({
    preset: PropTypes.oneOf(Object.keys(presets)),
    color: PropTypes.oneOf([...Object.keys(colors), 'inherit']),
    fontWeight: PropTypes.oneOf([500, 700, 900]),
    textAlign: PropTypes.oneOf(['left', 'center', 'right']),
    lineHeight: PropTypes.number,
    fontSize: PropTypes.number,
  }),
};

Text.defaultProps = {
  htmlTag: 'span',
};
