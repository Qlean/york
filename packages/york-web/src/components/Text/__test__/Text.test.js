import React from 'react';
import { shallow } from 'enzyme';
import Text from '../Text';

describe('<Text />', () => {
  test('should have text \'Test\'', () => {
    const text = shallow(<Text>Test</Text>);
    expect(text.render().text()).toEqual('Test');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have color #123456', () => {
    const text = shallow(<Text color="#123456">Test</Text>);
    expect(text).toHaveStyleRule('color', '#123456');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have green (#20A052) from york-core colors', () => {
    const text = shallow(<Text color="green">Test</Text>);
    expect(text).toHaveStyleRule('color', '#20A052');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have font-size 20px and line-height 30px', () => {
    const text = shallow(<Text fontSize={20} lineHeight={30}>Test</Text>);
    expect(text).toHaveStyleRule('font-size', '20px');
    expect(text).toHaveStyleRule('line-height', '30px');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have bold font', () => {
    const text = shallow(<Text fontWeight={700}>Test</Text>);
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have light font', () => {
    const text = shallow(<Text fontWeight={300}>Test</Text>);
    expect(text).toHaveStyleRule('font-weight', '300');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have default text2 preset', () => {
    const text = shallow(<Text>Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '500');
    expect(text).toHaveStyleRule('font-size', '16px');
    expect(text).toHaveStyleRule('line-height', '25px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have ph1 preset', () => {
    const text = shallow(<Text preset="ph1">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-size', '70px');
    expect(text).toHaveStyleRule('line-height', '75px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have ph2 preset', () => {
    const text = shallow(<Text preset="ph2">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-size', '60px');
    expect(text).toHaveStyleRule('line-height', '65px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have ph3 preset', () => {
    const text = shallow(<Text preset="ph3">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-size', '50px');
    expect(text).toHaveStyleRule('line-height', '55px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have h1 preset', () => {
    const text = shallow(<Text preset="h1">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-size', '40px');
    expect(text).toHaveStyleRule('line-height', '45px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have h2 preset', () => {
    const text = shallow(<Text preset="h2">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-size', '30px');
    expect(text).toHaveStyleRule('line-height', '35px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have h3 preset', () => {
    const text = shallow(<Text preset="h3">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-size', '25px');
    expect(text).toHaveStyleRule('line-height', '30px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have h4 preset', () => {
    const text = shallow(<Text preset="h4">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-size', '20px');
    expect(text).toHaveStyleRule('line-height', '25px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have text1 preset', () => {
    const text = shallow(<Text preset="text1">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '500');
    expect(text).toHaveStyleRule('font-size', '20px');
    expect(text).toHaveStyleRule('line-height', '30px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have link preset', () => {
    const text = shallow(<Text preset="link">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-size', '15px');
    expect(text).toHaveStyleRule('line-height', '25px');
    expect(text).toHaveStyleRule('text-transform', 'uppercase');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have caption1 preset', () => {
    const text = shallow(<Text preset="caption1">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '500');
    expect(text).toHaveStyleRule('font-size', '14px');
    expect(text).toHaveStyleRule('line-height', '20px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should have caption2 preset', () => {
    const text = shallow(<Text preset="caption2">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '500');
    expect(text).toHaveStyleRule('font-size', '12px');
    expect(text).toHaveStyleRule('line-height', '15px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should reassign h1 preset parameters', () => {
    const text = shallow(<Text fontSize={7} lineHeight={10} fontWeight={300} preset="h1">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '300');
    expect(text).toHaveStyleRule('font-size', '7px');
    expect(text).toHaveStyleRule('line-height', '10px');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
  test('should remove link preset uppercase transform', () => {
    const text = shallow(<Text preset="link" textTransform="none">Test</Text>);
    expect(text).toHaveStyleRule('color', '#222222');
    expect(text).toHaveStyleRule('font-weight', '700');
    expect(text).toHaveStyleRule('font-size', '15px');
    expect(text).toHaveStyleRule('line-height', '25px');
    expect(text).toHaveStyleRule('text-transform', 'none');
    expect(text).toHaveStyleRule('font-family', 'MuseoSansCyrl');
    expect(text.render()).toMatchSnapshot();
  });
});
