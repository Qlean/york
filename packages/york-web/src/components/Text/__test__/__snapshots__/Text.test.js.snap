// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should have bold font 1`] = `
.c0 {
  font-family: MuseoSansCyrl;
  font-weight: 700;
  font-size: 16px;
  line-height: 25px;
  color: #222222;
}

<span
  className="c0"
  color="coal"
  fontSize={16}
  fontWeight="bold"
>
  Test
</span>
`;

exports[`<Text /> should have color #123456 1`] = `
.c0 {
  font-family: MuseoSansCyrl;
  font-weight: 500;
  font-size: 16px;
  line-height: 25px;
  color: #123456;
}

<span
  className="c0"
  color="#123456"
  fontSize={16}
  fontWeight="normal"
>
  Test
</span>
`;

exports[`<Text /> should have font-size 20px and line-height 30px 1`] = `
.c0 {
  font-family: MuseoSansCyrl;
  font-weight: 500;
  font-size: 20px;
  line-height: 30px;
  color: #222222;
}

<span
  className="c0"
  color="coal"
  fontSize="20"
  fontWeight="normal"
>
  Test
</span>
`;

exports[`<Text /> should have green (#20A052) from york-core colors 1`] = `
.c0 {
  font-family: MuseoSansCyrl;
  font-weight: 500;
  font-size: 16px;
  line-height: 25px;
  color: #20A052;
}

<span
  className="c0"
  color="green"
  fontSize={16}
  fontWeight="normal"
>
  Test
</span>
`;

exports[`<Text /> should have light font 1`] = `
.c0 {
  font-family: MuseoSansCyrl;
  font-weight: 300;
  font-size: 16px;
  line-height: 25px;
  color: #222222;
}

<span
  className="c0"
  color="coal"
  fontSize={16}
  fontWeight="light"
>
  Test
</span>
`;

exports[`<Text /> should have text 'Test' 1`] = `
.c0 {
  font-family: MuseoSansCyrl;
  font-weight: 500;
  font-size: 16px;
  line-height: 25px;
  color: #222222;
}

<span
  className="c0"
  color="coal"
  fontSize={16}
  fontWeight="normal"
>
  Test
</span>
`;

exports[`<Text /> should match snapshot with default props 1`] = `
.c0 {
  font-family: MuseoSansCyrl;
  font-weight: 500;
  font-size: 16px;
  line-height: 25px;
  color: #222222;
}

<span
  className="c0"
  color="coal"
  fontSize={16}
  fontWeight="normal"
>
  Test
</span>
`;
